src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts

```diff
@@ -65,14 +65,11 @@ export const fetchEntity = async <
   try {
     const res = await fn(filters);
     queryClient.setQueryData(QUERY_KEY, res);
-
-    // If we reached the end of the list, we set the eof flag to prevent future requests.
-    if (!res.next) {
-      eofRef.current = { ...eofRef.current, [queryKeyString]: filters.page! };
-      // Eof is cached based, the same way, we cache the fetching data. Otherwise there would
-      // be request to non existing pages after reload.
-      queryClient.setQueryData(eofQueryKey, eofRef.current);
-    }
+    // Eof is cached based, the same way, we cache the fetching data. Otherwise there would
+    // be request to non existing pages after reload.
+    const totalPages = Math.ceil(res.count / perPage);
```

Comment:
Can you confirm that this change around removing `if(!res.next)` is purely to make the code simpler or it does solve an issue?

Reply:
I don't remember it fixing a bug. But i felt better having `eofRef.current` up to date all along on debuging and i saw no reason not to keep it like this at the end.

Reply:
Allright